client/client.go

```diff
@@ -122,25 +123,41 @@ func bind(cfg *Config, conn ldaputil.Connection) error {
 		return errors.New("unable to bind due to lack of configured password")
 	}
 
-	if cfg.BindDN == "" {
-		return errors.New("must provide binddn")
-	}
-
-	origErr := conn.Bind(cfg.BindDN, cfg.BindPassword)
-	if origErr == nil {
+	if cfg.UPNDomain != "" {
```

Comment:
This is copied from [client.go#L128-L155](https://github.com/hashicorp/vault-plugin-secrets-ad/blob/main/plugin/client/client.go#L128-L155) to maintain compatibility for users migrating from AD secrets to LDAP secrets.